### ISSUE TITLE
unusual behaviour for the findpeaks function

### ISSUE THREAD

**Issue by mfanourakis** [gjmvanboxtel/gsignal#20]
The `findpeaks` function crashes when there is a single peak in the data

To reproduce:
```R
findpeaks(c(1,2,3,4,5,4,3,2,1))
```
> Error in if (any(D) < MinPeakDistance) { : 
>  missing value where TRUE/FALSE needed

Next, the following 3 give inconsistent results:
```R
findpeaks(c(5,0,0,0,0,0,0,0,0,0))$loc
```
> NULL

```R
findpeaks(c(5,5,0,0,0,0,0,0,0,0))$loc
```
>[1] 1 2

```R
findpeaks(c(5,5,5,0,0,0,0,0,0,0))$loc
```
> Error in if (any(D) < MinPeakDistance) { : 
>  missing value where TRUE/FALSE needed

**Comment by mfanourakis** [comment]
Just noticed that the merged pull request #19  fixed the error, however the inconsistent results remain

**Comment by gjmvanboxtel** [comment]
Hi,

Thanks for your report. Indeed, the error has been fixed.

As for the inconsistencies, do you mean that findpeaks should find peaks at positions 1, 3 and 5? This is perhaps a matter of principle and I could see some arguments in favor of that. But neither Matlab nor Octave finds peak in those cases.

But perhaps you mean something else?
